geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/locator/GMSLocator.java

```diff
@@ -132,15 +130,20 @@ public GMSLocator(InetAddress bindAddress, String locatorString, boolean usePref
     this.objectDeserializer = objectDeserializer;
   }
 
-  public synchronized boolean setMembership(Membership<ID> membership) {
-    if (services == null || services.isStopped()) {
-      services = ((GMSMembership<ID>) membership).getServices();
-      localAddress = services.getMessenger().getMemberID();
+  /**
+   * Called initially and after each auto-reconnect. See restart handlers in InternalLocator
+   * up in geode-core. Services must be started before this call.
+   *
+   */
+  public synchronized boolean setServices(
```

Comment:
This method had and has important initialization logic (see new block comment above). This is the code that needs to be invoked after services are _start_ed but before the distributed system is joined.